src/cursor/abstract_cursor.ts

```diff
@@ -489,7 +493,7 @@ export abstract class AbstractCursor<
         await this.fetchBatch();
       } while (!this.isDead || (this.documents?.length ?? 0) !== 0);
     } finally {
-      if (this.cursorOptions.timeoutMode === CursorTimeoutMode.ITERATION && this.cursorId != null) {
+      if (this.cursorOptions.timeoutMode === CursorTimeoutMode.ITERATION) {
```

Comment:
Just curious - why are we removing the check for `cursorId != null`?

Reply:
Regardless of whether or not the cursor id is null after the fetch batch, we should still clear the timeout context since we would have initialized it in the `cursorInit` call.
I don't feel very strongly about this, so I can revert this if needed, but regardless of the path the cursor takes, clearing the timeoutContext at the end of the iteration is correct.

* uninitialized -> successfully initialized; timeoutContext is initialized, should be cleared
* uninitialized -> failed to initialize -> cleanup is called;  timeoutContext would have been cleared and set to undefined, but the nullish check makes this alright
* initialized -> successful getMore; timeout context should be cleared
* initialized -> failed getMore -> cleanup is called; same situation as failing to initialize